testsuite/integration/ws/src/test/java/org/jboss/as/test/integration/ws/wsse/trust/WSBearerElytronSecurityPropagationTestCase.java

```diff
@@ -0,0 +1,203 @@
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.jboss.shrinkwrap.api.exporter.ZipExporter;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TestRule;
+import org.junit.rules.TestWatcher;
+import org.junit.runner.Description;
+import org.junit.runner.RunWith;
+
+/**
+ * Test for WFLY-10480 with Elytron security domain
+ *
+ */
+@RunWith(Arquillian.class)
+@ServerSetup(WSTrustTestCaseElytronSecuritySetupTask.class)
+public class WSBearerElytronSecurityPropagationTestCase {
+    private static final String BEARER_STS_DEP = "jaxws-samples-wsse-policy-trust-sts-bearer";
+    private static final String BEARER_SERVER_DEP = "jaxws-samples-wsse-policy-trust-bearer";
+    @Rule
+    public TestRule watcher = new WrapThreadContextClassLoaderWatcher();
+    @ArquillianResource
+    private URL serviceURL;
+
+
+    @Deployment(name = BEARER_STS_DEP, testable = false)
+    public static WebArchive createBearerSTSDeployment() {
+        WebArchive archive = ShrinkWrap.create(WebArchive.class, BEARER_STS_DEP + ".war");
+        archive
+                .setManifest(new StringAsset("Manifest-Version: 1.0\n"
+                        + "Dependencies: org.jboss.ws.cxf.jbossws-cxf-client,org.jboss.ws.cxf.sts annotations\n"))
+                .addClass(org.jboss.as.test.integration.ws.wsse.trust.stsbearer.STSBearerCallbackHandler.class)
+                .addClass(org.jboss.as.test.integration.ws.wsse.trust.stsbearer.SampleSTSBearer.class)
+                .addClass(org.jboss.as.test.integration.ws.wsse.trust.shared.WSTrustAppUtils.class)
+                .addAsWebInfResource(createFilteredAsset("WEB-INF/wsdl/bearer-ws-trust-1.4-service.wsdl"), "wsdl/bearer-ws-trust-1.4-service.wsdl")
+                .addAsWebInfResource(WSTrustTestCase.class.getPackage(), "WEB-INF/stsstore.jks", "classes/stsstore.jks")
+                .addAsWebInfResource(WSTrustTestCase.class.getPackage(), "WEB-INF/stsKeystore.properties", "classes/stsKeystore.properties")
+                .addAsManifestResource(WSTrustTestCase.class.getPackage(), "META-INF/permissions.xml", "permissions.xml")
+                .setWebXML(WSTrustTestCase.class.getPackage(), "WEB-INF/bearer/web.xml");
+        return archive;
+    }
+
+    @Deployment(name = BEARER_SERVER_DEP, testable = false)
+    public static WebArchive createBearerServerDeployment() {
+        WebArchive archive = ShrinkWrap.create(WebArchive.class, BEARER_SERVER_DEP + ".war");
+        archive
+                .setManifest(new StringAsset("Manifest-Version: 1.0\n"
+                        //+ "Dependencies: org.jboss.ws.cxf.jbossws-cxf-client,org.apache.cxf.impl,org.apache.cxf.ws-security\n"))
+                        + "Dependencies: org.apache.cxf.impl,org.apache.cxf,org.jboss.ws.spi,org.jboss.ws.cxf.jbossws-cxf-server,"
+                        + "org.apache.ws.security,org.jboss.ws.jaxws-client\n"))
```

Comment:
honestly, I have a small concern with the `org.jboss.ws.cxf.jbossws-cxf-server` dependency. Is it really needed? What was missing when using the commented out dependencies line?

Reply:
I tried with just "Dependencies: org.jboss.ws.cxf.jbossws-cxf-client,org.apache.cxf.impl" and the test passes, 'org.apache.cxf.impl' is needed because of SamlSecurityContextInInterceptor using CXF specific classes.

Reply:
Thanks @jbliznak . Yes, the need for cxf impl is clear.